### ISSUE TITLE
Investigate swapchain present on webgpu

### ISSUE THREAD

**Issue by sagudev** [servo/servo#33679]
Currently we do swapchain present on `expire_current_texture` before destruction of texture which is enough for HTMLCanvasElement (but might not be for OffScreen canvas) as `expire_current_texture` is (well should per spec) be always called before presenting, but we would need to call this before bitmap read, this would match the spec https://www.w3.org/TR/webgpu/#abstract-opdef-get-a-copy-of-the-image-contents-of-a-context:
> Ensure that all submitted work items (e.g. queue submissions) have completed writing to the image (via context.[[[currentTexture]]](https://www.w3.org/TR/webgpu/#dom-gpucanvascontext-currenttexture-slot)).

Alternatively we could do swapchain present when command buffers do changes (write only?) to presentation texture (something like this is also done in mozilla, but they do it for all canvases on all submits unconditionally: https://searchfox.org/mozilla-central/rev/abeb2aa5df6f02de3c12c54ff7b0865c34c3fde9/dom/webgpu/CanvasContext.cpp#207).

TL;DR: We can copy texture to presentation buffer:

- after writes (to textures; can be done smarter than in mozilla but might not be worth it)
- before reads (of presentation buffers).

**Comment by sagudev** [comment]
for obtain pixel data we need to block on content timeline via waiting for IPC response rather than blocking in wgpu thread

**Comment by sagudev** [comment]
I think our doing this lazily is better, so we will continue to do so in https://github.com/servo/servo/pull/38717
